src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,91 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> String {
```

Comment:
Can you please switch this to -> Result<String> to let user handle the error as suggested by @tinaun, that makes this similar to other std::io parts, in the example, you could probably use ?.

Reply:
It took a little longer to implement, because i am not familiar with error handling.